lib/eosknowledgeprivate/ekn-media-bin.c

```diff
@@ -869,6 +869,12 @@ ekn_media_bin_dispose (GObject *object)
   EknMediaBin *self = EKN_MEDIA_BIN (object);
   EknMediaBinPrivate *priv = EMB_PRIVATE (self);
 
+  if (priv->timeout_id)
+    {
+      g_source_remove (priv->timeout_id);
+      priv->timeout_id = 0;
+    }
+
```

Comment:
We need to remove this from finalize() and while we are at it we could move it to a new inline function
ekn_media_bin_revealer_timeout_clear() and reuse it in ekn_media_bin_revealer_timeout()